server/acl/acl.go

```diff
@@ -0,0 +1,163 @@
+// Package acl provides a Go library for performing Access Control using
+// policy documents, similar to AWS IAM.
+//
+// In general, this package follows the same guidelines set forth by IAM when
+// evaluating IAM policies. See http://goo.gl/MY8C2r.
+package acl
+
+import (
+	"context"
+	"errors"
+	"strings"
+)
+
+type Effect int
+
+const (
+	Deny Effect = iota
+	Allow
+)
+
+// ACL is a composition of a Context, and the set of policy documents that
+// should be used to determine if the request should be allowed.
+type ACL struct {
+	Context  Context
+	Policies Policies
+}
+
+func (l *ACL) Allowed() bool {
+	return l.Policies.Allowed(l.Context)
+}
+
+// Context represents a compiled request context, which is provided to policies
+// to check if the request should be allowed.
+type Context struct {
+	// The Action that's being invoked.
+	Action string
+
+	// The Resource that the action is being invoked on.
+	Resource string
+}
+
+// Policy is used to define whether an action is allowed or denied.
+type Policy struct {
```

Comment:
I'm thinking that this will be expanded with something similar to the `Condition` key in IAM policy documents, so you could do things like allow/deny requests that didn't have 2fa enabled, limit deployments to a specific Docker repo, etc.